src/components/screens/forms/forms-data-screen.tsx

```diff
@@ -1,32 +1,229 @@
+import { EqualizerRounded, HistoryRounded } from "@mui/icons-material";
+import { useNavigate } from "react-router-dom";
+
+/**
+ * Interface for single form row
+ */
+interface Row {
+  id: string;
+  title: string;
+  slug?: string;
+  monthlyReplies: number,
+  replyViewDelay: moment.Duration
+}
 
 /**
  * Forms data screen component
  */
-const FormsDataScreen: React.FC = () => (
-  <>
-    <NavigationTabContainer>
-      <NavigationTab
-        text={ strings.navigationHeader.formsScreens.formScreen }
-        to="./.."
-      />
-      <NavigationTab
-        text={ strings.navigationHeader.formsScreens.formDataScreen }
+const FormsDataScreen: React.FC = () => {
+  const errorContext = useContext(ErrorContext);
+
+  const apiClient = useApiClient(Api.getApiClient);
+  const { metaformsApi, repliesApi, auditLogEntriesApi } = apiClient;
+  const [ rows, setRows ] = useState<Row[]>([]);
+  const [ loading, setLoading ] = useState(false);
+  const navigate = useNavigate();
+
+  /**
+   * Builds a row for the table
+   *
+   * @param form form
+   * @param replies replies
+   * @param auditLogEntries audit log entries
+   */
+  const buildRow = (form: Metaform, replies: Reply[], auditLogEntries: AuditLogEntry[]): Row => {
+    const monthlyAverageReplies = MetaformUtils.getMonthlyAverageReply(replies);
+    const averageReplyViewDelay = MetaformUtils.getAverageReplyViewDelay(auditLogEntries);
+
+    return {
+      id: form.id || "",
+      title: form.title || strings.formScreen.noTitle,
+      slug: form.slug,
+      monthlyReplies: monthlyAverageReplies,
+      replyViewDelay: averageReplyViewDelay
+    };
+  };
+
+  /**
+   * Loads forms
+   *
+   * @return metaform list promise
+   */
+  const loadForms = async (): Promise<Metaform[]> => {
+    try {
+      return await metaformsApi.listMetaforms({
+        memberRole: MetaformMemberRole.Manager
+      });
+    } catch (e) {
+      errorContext.setError(strings.errorHandling.adminFormsDataScreen.listForms, e);
+      return [];
+    }
+  };
+
+  /**
+   * Loads relies
+   *
+   * @param metaformId metaform id
+   * @return replies list promise
+   */
+  const loadReplies = async (metaformId: string): Promise<Reply[]> => {
+    try {
+      return await repliesApi.listReplies({ metaformId: metaformId });
+    } catch (e) {
+      errorContext.setError(strings.errorHandling.adminFormsDataScreen.listReplies, e);
+      return [];
+    }
+  };
+
+  /**
+   * Loads audit log entries
+   *
+   * @param metaformId metaform id
+   * @return audit log entries list promise
+   */
+  const loadAuditLogEntries = async (metaformId: string): Promise<AuditLogEntry[]> => {
+    try {
+      return await auditLogEntriesApi.listAuditLogEntries({ metaformId: metaformId });
+    } catch (e) {
+      errorContext.setError(strings.errorHandling.adminFormsDataScreen.listAuditLogEntries, e);
+      return [];
+    }
+  };
+
+  /**
+   * Loads data
+   */
+  const loadData = async () => {
```

Comment:
The body of this method should propably be wrapped with try catch, even tho errors are handled properly in loadReplies() and loadAuditLogEntries(). Atleast it minimizes the risk of unhandled promise rejection error.